### PR TITLE
[W08H03] Added big tree test and edited existing tests

### PR DESCRIPTION
**Pull request by HendrikHuebner**
edits: made parallelIteratorsTest seeded, asserted NoSuchElementException return message

### CODE REVIEW

File: w08h03/test/pgdp/datastructures/UnitTest.java [hunk 1]
```diff
@@ -97,7 +98,7 @@ private String generateString(Random rnd) {
 	public void parallelIteratorsTest() {
 		QuarternarySearchTree<Integer> t = new QuarternarySearchTree<>();
 
-		Random random = new Random();
+		Random random = new Random(seed);
```
Comment (nilsreichardt) [inline w08h03/test/pgdp/datastructures/UnitTest.java:101]: Thanks for adding the seed to `Random` 👍 However, can you revert this change because we are going to remove this test (see #174) and this would result a merge conflict.

File: w08h03/test/pgdp/datastructures/UnitTest.java [hunk 2]
```diff
@@ -39,7 +39,8 @@ <Type extends Comparable<Type>> void testArray(Type[] input) {
 	@Test
 	@DisplayName("should iterate over empty graph")
 	public void emptyTest() {
-		assertThrows(NoSuchElementException.class, () -> (new QuarternarySearchTree<Integer>()).iterator().next());
+		Exception e = assertThrows(NoSuchElementException.class, () -> (new QuarternarySearchTree<Integer>()).iterator().next());
+		assertEquals(e.getMessage(), "Trying to call next on an empty QuarternarySearchTreeIterator");
```
Comment (nilsreichardt) [inline w08h03/test/pgdp/datastructures/UnitTest.java:43]: Thanks for improving this ❤️ But for the next time it would be better to split this PR into two PRs, one for adding a new test, one for fixing an existing test 👍

**Review by nilsreichardt — APPROVED** [review]
LGTM (just the 1 note with the seed)

File: w08h03/test/pgdp/datastructures/UnitTest.java (hunk 1 above)
Comment (HendrikHuebner) [reply w08h03/test/pgdp/datastructures/UnitTest.java]: done

Comment (nilsreichardt) [reply w08h03/test/pgdp/datastructures/UnitTest.java]: thanks 👍

File: w08h03/test/pgdp/datastructures/UnitTest.java (hunk 2 above)
Comment (nilsreichardt) [reply w08h03/test/pgdp/datastructures/UnitTest.java:43]: Are you sure that the order of `expected` and `actual` is correct in `assertEquals`?

Comment (nilsreichardt) [reply w08h03/test/pgdp/datastructures/UnitTest.java:43]: Short question: Where is the message "Trying to call next on an empty QuarternarySearchTreeIterator" coming from? Because the test is testing that you throw like this `throw new NoSuchElementException("Trying to call next on an empty QuarternarySearchTreeIterator")`

Comment (HendrikHuebner) [reply w08h03/test/pgdp/datastructures/UnitTest.java:43]: There is a line in the template that you are supposed to use to throw a NoSuchElementException. I missed that too at first and didn't add an Error message at all so I decided to make a test for it.

Comment (nilsreichardt) [reply w08h03/test/pgdp/datastructures/UnitTest.java:43]: Oh, I see. Extremely good catch! Thanks 🙏

**Review by MaximilianAnzinger — COMMENTED** [review]
🐧
